Patch for Partner Stats 🏷

dlipeles opened the pull request:
## Patch for Partner Stats

![Screen Shot 2019-04-18 at 4 33 29 PM](https://user-images.githubusercontent.com/4713771/56389520-c3204880-61f7-11e9-97b1-34f63219f924.png)


#### Features

* Fix an issue partners were seeing with the stats on their dashboard. 
* Aggregate all stats for a referral code, instead of showing daily stats
* Add additional error handling for null or empty stats
* Ensure reporting works for partner generated campaigns/codes 

#### How To Test

1. As a `partner` create a campaign with promo codes 
2. As an `admin`, go to the referral promo page in the dashboard
3. Generate a report to update the stats
4. Ensure parity between partner stats, admin stats, and reports


#### Pull Request Checklist

- [ ] Adequate test coverage exists to prevent regressions -- [Guide to Testing](https://guides.rubyonrails.org/testing.html)
- [ ] XSS is mitigated -- [Guide to XSS Prevention](https://guides.rubyonrails.org/security.html#cross-site-scripting-xss)
- [ ] No raw SQL -- [Guide to SQL Injection Prevention](https://guides.rubyonrails.org/security.html#sql-injection)
- [ ] UI/UX is responsive -- [Guide to Responsive UI/UX](https://developers.google.com/web/fundamentals/design-and-ux/responsive/)
- [ ] Integrated Matomo for new UI elements -- [Guide to Matomo](https://developer.matomo.org/guides/integrate-introduction)
- [ ] Passes checklist for Progressive Web App -- [Guide to PWAs](https://developers.google.com/web/progressive-web-apps/checklist)

yachtcaptain23 commented on app/services/promo/registrations_stats_fetcher.rb:
```diff
@@ -62,6 +64,10 @@ def api_authorization_header
 
   def query_string(referral_codes)
-    "?" + referral_codes.map { |referral_code| "referral_code=#{referral_code}"}.join("&")
+    query_string = "?"
```
Preferably we use the older version since it's much more readable.

dlipeles replied:
Sounds good, fixed.

dlipeles commented:
@yachtcaptain23 Thanks for reviewing, to your point about error handling. I agree with you, normally we display the error to the user in a dialogue as seen here https://github.com/brave-intl/publishers/blob/a858d27daa9ffcb5734b818a409934ab30cdf4fb/app/javascript/views/referrals/referralsInformation/editCampaignDialog/EditCampaignDialog.tsx#L108

This particular case is a bit easier, we only need to handle empty stats gracefully (it will display as a 0 if no stats collected yet). No need to dialogue the user, so I think we are okay.

corymcdonald commented on app/javascript/views/referrals/referralsInformation/ReferralsInformation.tsx:
```diff
@@ -344,6 +348,19 @@ function ReferralsTable(props) {
   const rows = [];
 
   props.referralCodes.forEach((referralCode, index) => {
+    let downloads = 0;
```
Could you reuse the function you defined above?
`const { downloads, installs, thirtyDayUse } = processStats(referralCode)`

dlipeles replied:
Added to my backlog.

dlipeles commented:
Removed the `owner_promo_registrations_stats_job.rb` and its call from `sidekiq` until we implement polymorphic codes. We'll keep the `registrar` because it does some useful things like add the publisher `id` and the code `description`. 

I swept our production db and updated the few owner codes that did exist.
